pkg/yaml/types.go

```diff
@@ -0,0 +1,58 @@
+/*
```

Comment:
A bit confused, why do we need to have a separate directory for this? Why can we not make this a part of `pkg/spec/util.go`? Or, if we think this is more global that `pkg/spec`, then maybe create a root level `util` directory and put this there?

Reply:
because as we encounter more yaml types we can keep adding them here, I think what is done here makes sense!

Reply:
Sure, then maybe a `types` directory with our custom types

Reply:
Because it mainly uses yaml/json and it follows similar practices used in libcompose (although maybe we shouldn't use that as an example...)